### PR TITLE
Post the test results to codecov

### PR DESCRIPTION
**Pull request by LinuxSuRen**

### CODE REVIEW

**Comment by JohnNiang** [comment]
There has a test error in #1. Maybe we should fix it in this PR.

**Comment by JohnNiang** [comment]
I will fix it in another PR, but this PR looks good to me.

**Comment by JohnNiang** [comment]
hi @LinuxSuRen , you just need to execute command `go mod tidy` in your local environment, and commit those changes.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ jobs:
         run: |
           if [ -n "$(git status --porcelain)" ]; then
             echo 'To fix this check, run "go mod tidy"'
+            git diff
```
Comment (JohnNiang) [inline .github/workflows/main.yaml:25]: Please remove unused line for debug.

File: Makefile [hunk 2]
```diff
@@ -43,6 +43,7 @@ lint: ## Run golangci-lint against codes.
 	$(GOLANGCI_LINT) run
 
 test: fmt vet ## Run tests.
+	go mod tidy
```
Comment (JohnNiang) [inline Makefile:46]: I suggest we do not place command `go mod tidy` here.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (LinuxSuRen) [reply .github/workflows/main.yaml:25]: I don't think it's unused. It's useful to see the diff.

File: Makefile (hunk 2 above)
Comment (LinuxSuRen) [reply Makefile:46]: Why not?

File: .github/workflows/main.yaml (hunk 1 above)
Comment (JohnNiang) [reply .github/workflows/main.yaml:25]: > To fix this check, run "go mod tidy"

The tip shows how to fix this issue.

**Comment by LinuxSuRen** [comment]
I have no idea that why I cannot set up this project in Codecov. So, I'm going to split this PR into two.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (LinuxSuRen) [reply .github/workflows/main.yaml]: > The tip shows how to fix this issue.

In my case, the CI and local environments are different. It takes a lot of my time to find out it.

Anyway, it's very clear to print the diff. We don't need to run that command again.

File: Makefile (hunk 2 above)
Comment (JohnNiang) [reply Makefile]: I think it's a dependency issue and not affect test process.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -1,9 +1,12 @@
 name: Workflow for Jenkins Client
 
 on:
-  - push
-  - pull_request
-  - release
+  push:
+    branches:
+      - main
```
Comment (JohnNiang) [inline .github/workflows/main.yaml:6]: IMO, we could trigger `lint` and `test` workflows in every push instead of `main` branch only.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (LinuxSuRen) [reply Makefile]: Everyone should to check the following list:

* code format
* go mod
* vet
* others

So, the test is not just about the unit test. And `fmt` and `vet` are also not be part of the test process.

File: .github/workflows/main.yaml (hunk 3 above)
Comment (LinuxSuRen) [reply .github/workflows/main.yaml:6]: We don't have other branches for now.

**Comment by JohnNiang** [comment]
I still reserve my opinion.

/lgtm

**Pull request merged**